doc/source/whatsnew/v0.20.2.txt

```diff
@@ -83,6 +83,7 @@ Reshaping
 - Bug in ``DataFrame.stack`` with unsorted levels in MultiIndex columns (:issue:`16323`)
 - Bug in ``pd.wide_to_long()`` where no error was raised when ``i`` was not a unique identifier (:issue:`16382`)
 - Bug in ``Series.isin(..)`` with a list of tuples (:issue:`16394`)
+- Bug in ``DataFrame.agg`` and ``Series.agg`` with aggregating on non-callable attributes (:issue:`16404`)
```

Comment:
As it works right now, you really can use *any* attribute in the aggregation, so something like:
```python
In [20]: df
Out[20]:
     A    B     C
0  NaN  1.0   foo
1  2.0  NaN  None
2  3.0  3.0   bar

In [21]: df.agg({'A': ['count', 'size', 'name'],
    ...:         'B': ['count', 'size', 'name', 'data'],
    ...:         'C': 'size'})
    ...:
Out[21]:
         A                B    C
count    2                2  NaN
data   NaN  [1.0, nan, 3.0]  NaN
name     A                B  NaN
size     3                3  3.0
```
works. 

If it should only affect size, we could change the check in base.py to just see if the arg passed is `'size'`.